src/formatDate.js

```diff
@@ -8,7 +8,26 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldDateFormat = date.split(fromFormat[fromFormat.length - 1]);
+  const newDateFormat = [];
+
+  for (let i = 0; i < oldDateFormat.length; i++) {
+    if (toFormat.indexOf(fromFormat[i]) === -1) {
+      if (fromFormat[i].length === 2) {
+        if (+oldDateFormat[i] < 30) {
+          newDateFormat[toFormat.indexOf('YYYY')] = '20' + oldDateFormat[i];
+        } else {
+          newDateFormat[toFormat.indexOf('YYYY')] = '19' + oldDateFormat[i];
+        }
+      } else {
+        newDateFormat[toFormat.indexOf('YY')] = oldDateFormat[i].slice(-2);
+      }
+      continue;
+    }
+    newDateFormat[toFormat.indexOf(fromFormat[i])] = oldDateFormat[i];
+  }
+
+  return newDateFormat.join(toFormat[toFormat.length - 1]);
```

Comment:
The code assumes that the last element of `toFormat` is the separator, which is correct based on the task description. However, it would be more robust to explicitly check for the separator element in `toFormat` rather than relying on its position.